functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobSpec.groovy

```diff
@@ -1,32 +1,208 @@
-            entries == [
-                    'hello there',
+        entries == ['hello there',
                     'option opt1: testvalue',
                     'option opt1: testvalue',
-                     "node: $node",
+                    "node: $node",
                     'option opt2: a',
                     'this is script 2, opt1 is testvalue',
                     'hello there',
-                    'this is script 1, opt1 is testvalue',
-            ]
+                    'this is script 1, opt1 is testvalue',]
+    }
+
+    def "Create a job with multiple steps"() {
+        given:
+        def jobName = UUID.randomUUID().toString()
+        def jobXml = """
+        <joblist>
+           <job>
+              <name>${jobName}</name>
+              <group>api-test</group>
+              <description></description>
+              <loglevel>INFO</loglevel>
+              <multipleExecutions>true</multipleExecutions>
+              <dispatch>
+                <threadcount>1</threadcount>
+                <keepgoing>true</keepgoing>
+              </dispatch>
+              <sequence>
+                <command>
+                  <exec>echo 0</exec>
+                </command>
+                <command>
+                  <exec>echo 1</exec>
+                </command>
+              </sequence>
+           </job>
+        </joblist>"""
+
+        when:
+        def response = JobUtils.createJob(PROJECT_NAME, jobXml, client)
+
+        then:
+        response.successful
+        def jr = MAPPER.readValue(response.body().string(), CreateJobResponse.class)
+        def jobDetails = JobUtils.getJobDetailsById(jr.getSucceeded().get(0).id, MAPPER, client)
+        jobDetails.sequence.commands[0].exec == "echo 0"
+        jobDetails.sequence.commands[1].exec == "echo 1"
+    }
+
+    def "Create a job with a job schedule enabled"() {
+        given:
+        def jobName = UUID.randomUUID().toString()
+        def jobXml = """
+        <joblist>
+           <job>
+              <name>${jobName}</name>
+              <group>api-test</group>
+              <description></description>
+              <loglevel>INFO</loglevel>
+              <multipleExecutions>true</multipleExecutions>
+              <dispatch>
+                <threadcount>1</threadcount>
+                <keepgoing>true</keepgoing>
+              </dispatch>
+              <scheduleEnabled>true</scheduleEnabled>
+              <sequence>
+                <command>
+                  <exec>echo 0</exec>
+                </command>
+              </sequence>
+           </job>
+        </joblist>"""
+
+        when:
+        def response = JobUtils.createJob(PROJECT_NAME, jobXml, client)
+
+        then:
+        response.successful
+        def jr = MAPPER.readValue(response.body().string(), CreateJobResponse.class)
+        def jobDetails = JobUtils.getJobDetailsById(jr.getSucceeded().get(0).id, MAPPER, client)
+        jobDetails.scheduleEnabled == true
+    }
```

Comment:
question: It seems like there might be some tests in the JobExecutionSpec that also check if a job is being created with schedule/execution enabled/disabled (the tests starting with `test-job-flip-` name).

Would it be worth moving the tests added to JobSpec there and reorganizing the suite, or removing the tests added here as it seems that there's an overlap?

Reply:
Thanks for pointing this out. I was able to make a few minor changes to the tests in `JobExecutionSpec` and eliminate a few from this suite.